components/DeleteComponent/index.tsx

```diff
@@ -0,0 +1,47 @@
+import * as AlertDialogPrimitive from '@radix-ui/react-alert-dialog';
+import { ReactElement } from 'react';
+import { ButtonsContainer, Content, StyledContent, StyledDescription, StyledOverlay, TriggerButton,  } from './styles';
+import Garbage from '../../public/garbage.svg';
+import ColorfulGarbage from '../../public/colorfulGarbage.svg';
+import { Button } from '../Button';
+
+type ContentType = {
+    children: ReactElement;
+}
+
+function AlertDialogContent({ children, ...props }: ContentType) {
+  return (
+    <AlertDialogPrimitive.Portal>
+      <StyledOverlay />
+      <StyledContent {...props}>{children}</StyledContent>
+    </AlertDialogPrimitive.Portal>
+  );
+}
+
+export default function DeleteComponent () {
+    return (
+      <AlertDialogPrimitive.Root>
+        <AlertDialogPrimitive.Trigger asChild>
+          <TriggerButton>
+            <Garbage />
+          </TriggerButton>
+        </AlertDialogPrimitive.Trigger>
+        <AlertDialogContent data-testid='dialog'>
+          <Content>
+            <ColorfulGarbage />
+            <StyledDescription>
+              Quer excluir suas aulas?
+            </StyledDescription>
+            <ButtonsContainer>
+              <AlertDialogPrimitive.Action asChild>
+                <Button variant='ghost'>Não</Button>
+              </AlertDialogPrimitive.Action>
+              <AlertDialogPrimitive.Cancel asChild>
+                <Button variant='filled'>Com certeza</Button>
+              </AlertDialogPrimitive.Cancel>
+            </ButtonsContainer>
+          </Content>
+        </AlertDialogContent>
+      </AlertDialogPrimitive.Root>
```

Comment:
Confesso que acho essa estrutura muito complexa para fazer o que precisa ser feito. Aqui talvez valesse a pena criar um outro componente que isola a complexidade do radix e você usar esse componente no seu código. Um componente que você passa algumas props e pronto. Assim você não espalha as coisas do radix por todo o seu código.

Reply:
@wandersonmaceds   achei que isso é o que estava sendo feito nesse arquivo.. quando for usar o componente eu algum lugar vou usar o que eu criei e não o do radix. Confesso que não peguei mto a ideia aqui sobre o que deveria fazer.. ;/

Reply:
Faz sentido o que você diz, eu acho que tem um ganho você isolar ainda mais, por exemplo, seu componente `DeleteComponent`, poderia ter o seguinte corpo, imagina só:

```
<Dialog config={} cancel={}>
    <ColorfulGarbage />
    <StyledDescription>
        Quer excluir suas aulas?
    </StyledDescription>
</Dialog>
```

E esse componente Dialog, poderia isolar as coisas do radix. Inclusive, ele poderia ser aproveitado em qualquer outra situação que precise de um modal parecido, mas com ícone diferente, inclusive, customizando cores, etc via props.

Mas você pode deixar assim se preferir. :)

Reply:
hmmm.. dei uma refatorada aqui. Com base no que vc disse, minha ideia foi isolar tudo do radix e ter componentes genéricos para usar em outras situações. Assim eu criei um "Modal trigger" que ao chamar nós dizemos para ele quem vai ser o botão, e o  "modal content", onde ao chamar vamos especificar o que vai ali dentro do dialog.
Assim o componente de deleção que hj temos na aplicação só chama esses dois caras, passando as especificidades desse caso. Veja o que achou @wandersonmaceds

Reply:
Eu gostei, o DeleteComponent ficou bem mais simples. O que eu não entendi bem é: consigo usar só o trigger ou só o modal separadamente? Caso não, ter dois componentes não vai fazer muito sentido, visto que os dois parecem ter cara de um só. O trigger no caso, poderia ser uma prop do próprio Modal.

Reply:
É, tem razão.. o trigger até poderia ser usado com outro content, mas melhor assim. Transformei em um só